### PR TITLE
Fix: missing Handled errors from ASP.NET Core.

### PR DESCRIPTION
**Pull request by lucas-zimerman**
Set the handled mechanism to false from errors captured on SentryMiddleware.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### Fixes
+
+- ASP.NET Core: fix handled not being set for Handled exceptions ([]())
```
Comment (lucas-zimerman) [inline CHANGELOG.md:7]:
```suggestion
- ASP.NET Core: fix handled not being set for Handled exceptions ([#1111](https://github.com/getsentry/sentry-dotnet/pull/1111))

```

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,11 @@
 
 ## Unreleased
 
+### Fixes
+
+- ASP.NET Core: fix handled not being set for Handled exceptions ([#1111](https://github.com/getsentry/sentry-dotnet/pull/1111))
+
+
```
Comment (Tyrrrz) [inline CHANGELOG.md:9]: Redundant new line

```suggestion
### Fixes

- ASP.NET Core: fix handled not being set for Handled exceptions ([#1111](https://github.com/getsentry/sentry-dotnet/pull/1111))

```

**Review by Tyrrrz — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (lucas-zimerman) [reply CHANGELOG.md:9]: ty Tyrrrz, for some reason, my git is ignoring this kind of change O_o.

File: src/Sentry.AspNetCore/SentryMiddleware.cs [hunk 3]
```diff
@@ -127,6 +127,9 @@ public async Task InvokeAsync(HttpContext context)
 
                 void CaptureException(Exception e)
                 {
+                    e.Data[Mechanism.HandledKey] = false;
+                    e.Data[Mechanism.MechanismKey] = "SentryMiddleware";
```
Comment (bruno-garcia) [inline src/Sentry.AspNetCore/SentryMiddleware.cs:131]: Can we please differentiate between errors coming from unhandled exceptions and errors coming from `IExceptionHandlerFeature` ?

image

**Review by bruno-garcia — CHANGES_REQUESTED** [review]

File: src/Sentry.AspNetCore/SentryMiddleware.cs (hunk 3 above)
Comment (bruno-garcia) [reply src/Sentry.AspNetCore/SentryMiddleware.cs:131]: Maybe `Middleware.UnhandledException` (does it fit in the UI?)
And `IExceptionHandlerFeature`

Comment (lucas-zimerman) [reply src/Sentry.AspNetCore/SentryMiddleware.cs:131]: why not SentryMiddleware.UnhandledException ?

Comment (lucas-zimerman) [reply src/Sentry.AspNetCore/SentryMiddleware.cs:131]: it fits the UI (simulated crash)
image

Comment (lucas-zimerman) [reply src/Sentry.AspNetCore/SentryMiddleware.cs]: Set the Mechanism to "IExceptionHandlerFeature" and "SentryMiddleware.UnhandledException".
If the names are ok we are good to go :D

**Review by bruno-garcia — APPROVED** [review]
Perfect! thank you

**Pull request merged**